src/sieve.py

```diff
@@ -14,7 +14,19 @@ def sieve(n: int) -> list[int]:
     assert n > 0
     candidates = list(range(2, n + 1))
     primes = []
-
+    not_primes = []
+    
+    for i in candidates:
+        if i in candidates:
+            p = i
+            primes.append(i)
+            for j in candidates:
+                if j % p == 0 and j != p:
+                    candidates.remove(j)
```

Comment:
For reasons I haven't explained (yet) removing from a list is an expensive operation. You will get a faster solution if you build a new list instead, containing the candidates you want to keep:

```python
candidates = [cand for cand in candidates if cand % p != 0]
```
If you do this, you cannot run through the candidates with `for i in candidates`, but you can loop until there are none left, `while candidates`, and you can always get the next prime at `candidates[0]`.